File: fixtures/cpp/Animal.cpp
#include "Animal.h"
#include <iostream>

Animal::Animal(const std::string& name, int age) 
    : name(name), age(age) {}

Animal::~Animal() {}

void Animal::makeSound() const {
    std::cout << "Some generic animal sound" << std::endl;
}

void Animal::eat(const std::string& food) {
    std::cout << name << " is eating " << food << std::endl;
}

std::string Animal::getName() const {
    return name;
}

int Animal::getAge() const {
    return age;
}

File: fixtures/typescript/src/animal.ts
/**
 * 动物基类
 */
export class Animal {
    protected name: string;
    protected age: number;
    
    constructor(name: string, age: number) {
        this.name = name;
        this.age = age;
    }
    
    makeSound(): void {
        console.log('Some generic animal sound');
    }
    
    eat(food: string): void {
        console.log(`${this.name} is eating ${food}`);
    }
    
    getName(): string {
        return this.name;
    }
    
    getAge(): number {
        return this.age;
    }
    
    toString(): string {
        return `Animal(name=${this.name}, age=${this.age})`;
    }
}
